### PR TITLE
Fix #44: allow more than one try to fill username and password

### LINKED ISSUE #44: User should have more than one try to fill username and password
If I use registration plugin without saved ceredentials (in Vagrantfile), I have just one try to fill username and password correctly and if I make error, I have to do "vagrant halt && vagrant up".
It would be nice if this plugin support one of this options:
1) allow for example 3 attempts to fill credentials
2) give a hint how to do it manually after "vagrant ssh"

### PR DESCRIPTION
**Pull request by budhrg**
Fix #44 

Patched gem to try:- https://www.dropbox.com/s/dahxpg8is07wbwm/vagrant-registration-1.2.4.dev.gem 

Also displaying "Registration successful." message on successful registration.

### CODE REVIEW

**Comment by naina-verma** [comment]
LGMT!

File: README.md [hunk 1]
```diff
@@ -115,7 +115,7 @@ Vagrant.configure('2') do |config|
 end
 ```
 
-If you do not provide credentials, you will be prompted for them in the "up process."
+If you do not provide credentials, you will be prompted for them with maximum three attempts during the "vagrant up" process.
```
Comment (budhrg) [inline README.md]: @Preeticp , does this line make sense for you?

File: lib/vagrant-registration/action/register.rb [hunk 2]
```diff
@@ -5,34 +5,36 @@ module Registration
...
 
               # Offer to register ATM or skip
-              register_now = env[:ui].ask I18n.t('registration.action.register.prompt')
+              register_now = ui.ask I18n.t('registration.action.register.prompt')
 
               if register_now == 'n'
                 config.skip = true
               else
-                config = register_on_screen(machine, env[:ui])
+                config = process_registration(guest, machine, ui, config)
```
Comment (LalatenduMohanty) [inline lib/vagrant-registration/action/register.rb]: @budhrg method `register_on_screen` was only used here. Now that you have used a new method `process_registration` the old method is an unused code. So why not re-factor `register_on_screen` rather implementing a new method?

Comment (budhrg) [inline lib/vagrant-registration/action/register.rb]: @LalatenduMohanty `register_on_screen` is being consumed by `process_registration` now. :smile:

Comment (LalatenduMohanty) [inline lib/vagrant-registration/action/register.rb]: @budhrg Oops! my bad :)

**Comment by LalatenduMohanty** [comment]
@budhrg can you please rebase?

File: README.md (hunk 1 above)
Comment (Preeticp) [inline README.md]: @budhrg how about: You will receive a maximum of 3 prompts for the credentials, if you do not provide them during the ``Vagrant up`` process.

**Comment by budhrg** [comment]
@LalatenduMohanty Waiting for preeti to comment on doc side. Then I will rebase.

**Comment by LalatenduMohanty** [comment]
LGTM

**Comment by LalatenduMohanty** [comment]
@budhrg you should add acceptance test for this PR.

**Comment by budhrg** [comment]
Working on acceptance test now.

**Comment by budhrg** [comment]
@LalatenduMohanty Create issue https://github.com/projectatomic/adb-vagrant-registration/issues/106. 

Could you verify and merge it?

**Comment by LalatenduMohanty** [comment]
LGTM.

**Comment by LalatenduMohanty** [comment]
merged

**Pull request closed without merging**